onnxruntime/core/providers/rknpu/shaper.h

```diff
@@ -0,0 +1,135 @@
+                    const std::string& output_name);
+  void Gather(const std::string& input_name,
+              const std::string& indices_name,
+              const int32_t axis,
+              const std::string& output_name);
+  void Pool(const std::string& input_name, int32_t padding_left,
+            int32_t padding_right, int32_t padding_top,
+            int32_t padding_bottom, int32_t stride_x, int32_t stride_y,
+            int32_t width, int32_t height, const std::string& output_name);
+  void Pool(const std::string& input_name,
+            const std::vector<int32_t> kernel_shape,
+            const std::vector<int32_t> pads,
+            const std::vector<int32_t> strides,
+            const std::string& output_name);
+  void Softmax(const std::string& input_name, const std::string& output_name);
+  void Relu(const std::string& input_name, const std::string& output_name);
+  void Concat(const std::vector<std::string>& input_names, uint32_t axis,
+              const std::string& output_name);
+  void LRN(const std::string& input_name, const std::string& output_name);
+  void FC(const std::string& input_name, const std::string& weight_name,
+          const std::string& output_name);
+  void Eltwise(const std::string& input1_name, const std::string& input2_name,
+               const std::string& output_name);
+  void Eltwise(const std::string& input1_name,
+               const std::string& output_name);
+  void Affine(const std::string& input_name, const std::string& output_name);
+  void Affine(const std::string& input_name, const std::string& a,
+              const std::string& b, const std::string& output_name);
+  void Identity(const std::string& input_name,
+                const std::string& output_name);
+  void BatchToSpace(const std::string& input_name,
+                    const std::vector<int32_t>& block_sizes,
+                    const std::string& output_name);
+  void SpaceToBatch(const std::string& input_name,
+                    const std::vector<int32_t>& block_sizes,
+                    const std::vector<int32_t>& pads,
+                    const std::string& output_name);
+  void BatchNorm(const std::string& input_name,
+                 const std::string& output_name);
+  void Reshape(const std::string& input_name,
+               const std::vector<int32_t>& shape,
+               const std::string& output_name);
+  void Transpose(const std::string& input_name, const std::vector<int32_t> perm,
+                 const std::string& output_name);
+  void Squeeze(const std::string& input_name, const std::vector<int32_t>& axes,
+               const std::string& output_name);
+  void Unsqueeze(const std::string& input_name,
+                 const std::vector<int32_t>& axes,
+                 const std::string& output_name);
+  void AddShape(const std::string& name, const Shape& shape);
+  size_t GetSize(const std::string& name);
+  void Clear();
+
+  inline const Shape& operator[](const std::string& key) {
```

Comment:
>r[] [](start = 29, length = 3)

Semantical mismatch between interface and impl. [] usually does not check the bounds and doesnot throw. However, impl employes at() which does the opposite.

Reply:
is these correct?
```
  inline const Shape& operator[](const std::string& key) {
    return shape_map_[key];
  }
```